src/SymbolTable.ts

```diff
@@ -12,6 +12,8 @@ export class SymbolTable {
 
     private symbols = new Map<string, BscSymbol[]>();
 
+    public readonly children: SymbolTable[] = [];
```

Comment:
I don't think I like this `children` concept. I know it makes things slightly easier during cleanup. However, every a user types in a .brs file, the previous `BrsFile` is destroyed and a brand new `BrsFile` is created. Which means that we could potentially hold on to those old references for a while in these scopes. 

I think instead, the scope should wire itself up to all children at the start of `validate()` and then unwire at the end of `validate()`.  I believe currently the "children" would persist after the validate is finished.